packages/config-loader/src/lib/reader.ts

```diff
@@ -68,6 +74,17 @@ export async function readConfigFile(
       }
     }
 
+    const [secretKey] = Object.keys(obj).filter(key => key.startsWith('$'));
+    if (secretKey) {
+      try {
+        return await ctx.readSecret(path, {
+          [secretKey.slice(1)]: obj[secretKey],
+        });
```

Comment:
Hm! Why is this last argument an object?

Reply:
Easy backwards compatibility until we remove the `$secret` way of doing it